test/WPow.t.sol

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import "ds-test/test.sol";
+
+import "./WPow.sol";
+
+contract FixedPointMathLibTest is DSTest {
+    // test a couple of concrete cases
+    function testWPow() public {
+        assertEq(WPow.wpow(2e18, 2), 4e18);
+        assertEq(WPow.wpow(2e18, 4), 16e18);
+
+        assertEq(WPow.wpow(0, 0), 1e18);
+        assertEq(WPow.wpow(0, 1), 0);
+        assertEq(WPow.wpow(0, 10), 0);
+
+        assertEq(WPow.wpow(1, 0), 1e18);
+        assertEq(WPow.wpow(1e18, 0), 1e18);
+    }
+
+    // helper method
+    function testWPowImpl(uint256 x, uint256 n) internal {
+        uint256 expected = 1e18;
+        if (n > 0) {
+            expected = x;
+        }
+        bool expectExpectedToIncrease = (x > 1e18);
+        if (n > 1) {
+            for (uint256 i = 1; i < n; ++i) {
+                unchecked {
+                    uint256 old_expected = expected;
+                    expected = (expected * x) / 1e18;
+                    // check for overflow if x > 1
+                    if (expectExpectedToIncrease && expected < old_expected) {
+                        return;
+                    }
+                    if (expected == old_expected) {
+                        break;
```

Comment:
I guess that we get out of the loop because we have reached the precision limits.